### PR TITLE
Announcing Rust 1.74.0

### PR DESCRIPTION
**Pull request by cuviper**

### CODE REVIEW

File: posts/2023-11-16-Rust-1.74.0.md [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+- [`impl<T: Clone, const N: usize> From<&[T; N]> for Vec<T>`](https://doc.rust-lang.org/stable/std/vec/struct.Vec.html#impl-From%3C%26%5BT;+N%5D%3E-for-Vec%3CT,+Global%3E)
+- [`impl<T: Clone, const N: usize> From<&mut [T; N]> for Vec<T>`](https://doc.rust-lang.org/stable/std/vec/struct.Vec.html#impl-From%3C%26mut+%5BT;+N%5D%3E-for-Vec%3CT,+Global%3E)
+- [`impl<T, const N: usize> From<[T; N]> for Arc<[T]>`](https://doc.rust-lang.org/stable/std/sync/struct.Arc.html#impl-From%3C%5BT;+N%5D%3E-for-Arc%3C%5BT%5D,+Global%3E)
+- [`impl<T, const N: usize> From<[T; N]> for Rc<[T]>`](https://doc.rust-lang.org/stable/std/rc/struct.Rc.html#impl-From%3C%5BT;+N%5D%3E-for-Rc%3C%5BT%5D,+Global%3E)
+
+These APIs are now stable in const contexts:
+
+- [`core::mem::transmute_copy`](https://doc.rust-lang.org/beta/std/mem/fn.transmute_copy.html)
+- [`str::is_ascii`](https://doc.rust-lang.org/beta/std/primitive.str.html#method.is_ascii)
+- [`[u8]::is_ascii`](https://doc.rust-lang.org/beta/std/primitive.slice.html#method.is_ascii)
```
Comment (Mark-Simulacrum) [inline posts/2023-11-16-Rust-1.74.0.md:119]:
```suggestion
- [`core::mem::transmute_copy`](https://doc.rust-lang.org/stable/std/mem/fn.transmute_copy.html)
- [`str::is_ascii`](https://doc.rust-lang.org/stable/std/primitive.str.html#method.is_ascii)
- [`[u8]::is_ascii`](https://doc.rust-lang.org/stable/std/primitive.slice.html#method.is_ascii)
```

File: posts/2023-11-16-Rust-1.74.0.md [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+## What's in 1.74.0 stable
+
+### Lint configuration through Cargo
+
+As proposed in [RFC 3389](https://rust-lang.github.io/rfcs/3389-manifest-lint.html),
+the `Cargo.toml` manifest now supports a `[lints]` table to configure the
+reporting level (forbid, deny, warn, allow) for lints from the compiler and
+other tools. So rather than setting `RUSTFLAGS` with `-F`/`-D`/`-W`/`-A`, which
+would affect the entire build, or using crate-level attributes like:
```
Comment (shepmaster) [inline posts/2023-11-16-Rust-1.74.0.md:28]: I don’t know which is right, but I’m hoping someone looked up if this is supposed to be affect or effect…

File: posts/2023-11-16-Rust-1.74.0.md [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+Cargo reference manual.
+
+[`lints`]: https://doc.rust-lang.org/stable/cargo/reference/manifest.html#the-lints-section
+[`workspace.lints`]: https://doc.rust-lang.org/stable/cargo/reference/workspaces.html#the-lints-table
+
+### Projections in opaque return types
+
+If you have ever received the error that a "return type cannot contain a
+projection or `Self` that references lifetimes from a parent scope," you may
+now rest easy! The compiler does now allow allows mentioning `Self` and
```
Comment (ehuss) [inline posts/2023-11-16-Rust-1.74.0.md:59]:
```suggestion
now rest easy! The compiler now allows mentioning `Self` and
```

File: posts/2023-11-16-Rust-1.74.0.md (hunk 2 above)
Comment (cuviper) [reply posts/2023-11-16-Rust-1.74.0.md:28]: I think I got it right, as a verb like "change" or "act on".
https://www.merriam-webster.com/grammar/affect-vs-effect-usage-difference

File: posts/2023-11-16-Rust-1.74.0.md [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+associated types in opaque return types, like `async fn` and `-> impl Trait`.
+This is the kind of feature that gets Rust closer to how you might just
+_expect_ it to work, even if you have no idea about jargon like "projection."
```
Comment (lcnr) [inline posts/2023-11-16-Rust-1.74.0.md:62]:
```suggestion
_expect_ it to work, even if you have no idea about jargon like "projection".
```

**Comment by cuviper** [comment]
Added a section from @arlosi on [zulip](https://rust-lang.zulipchat.com/#narrow/stream/241545-t-release/topic/1.2E74.2E0.20release/near/402148551) about cargo authentication.

**Review by Mark-Simulacrum — APPROVED** [review]

**Pull request merged**